Create .json files for other TAS diagram configurations and add to TAS class

sarahshi opened the pull request:
…S class

<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

morganjwilliams commented:
Hey @sarahshi! 

Apologies for the delay in getting back to this one. I'll check it out and see how it works locally. I'll also update the PR to point to `pyrolite/develop` instead of `pyrolite/main`.

Cheers!

morganjwilliams commented:
Looks like there are some minor conflicts due to working from `main` instead of `develop` as a base for the branch (seems to be contained to a docstring which was updated with Middlemost on `develop`), we can address that before we merge it across (essentially just need to merge the change from develop and the changes you've added).

sarahshi commented:
Hi @morganjwilliams. Awesome, apologies for the mistaken development branch. Let me know how I can help further!

morganjwilliams reviewed (changes requested):
Hey @sarahshi, a few notes below following up about this after I played around a little offline - looks good from a first pass, and I don't think any additional tests are needed here.

![TASVariants](https://user-images.githubusercontent.com/11415019/206996717-0d1de469-d94e-4bac-8269-c548a75bf670.png)

* I've added a few comments around the referencing - I think the Middlemost reference should be Middlemost (1994), which can be copied from `develop`.
* For now we'll work with the multiple-files-in-one-folder way of doing variants of plot templates, but I'll put some more thought into this for later releases and perhaps put together a bit of a guide to 'making plot templates and variants' as a tutorial.
* Please remove the `.DS_Store` binary files added to the repo (I suspect this is a Mac thing?)

For the merge conflict, we can either fix that here in GitHub's online tool or you could fix it locally and push to your branch (pretty much by copy-pasting the version of the docstring from develop).

sarahshi commented on pyrolite/util/classification.py:
```diff
@@ -351,15 +351,33 @@ class TAS(PolygonClassifier):
+    .. [#ref_1] Middlemost, Eric A. K. (1989).
+            Iron Oxidation Ratios, Norms and the
+            Classification of Volcanic Rocks. Chemical Geology 77, 1: 19–26.
+            https://doi.org/10.1016/0009-2541(89)90011-9.
```
Corrected, apologies for the mistaken reference. I've added the correction and provided the reference IDs for Middlemost (1994) and Le Maitre (2002) in both the TAS and QAP clases.

sarahshi commented on pyrolite/util/classification.py:
```diff
@@ -351,15 +351,33 @@ class TAS(PolygonClassifier):
     -----------
-    .. [#ref_1] Le Bas, M.J., Le Maitre, R.W., Woolley, A.R. (1992).
+    .. 
+    .. [#ref_1] Middlemost, Eric A. K. (1989).
```
Corrected!

morganjwilliams replied:
Thanks! I think there's a redundant `..` left here which can be removed but otherwise the changes work and this has fixed the conflict.

sarahshi replied:
Removed!

sarahshi commented:
Hi Morgan, thank you for reviewing! I believe I have corrected the code appropriately and removed the .DS_Store file (indeed, a remnant of Mac). I hope that merge conflict has been resolved!

morganjwilliams commented:
There's a few more spurious `.DS_Store` files about (can see where they are in the [files](https://github.com/morganjwilliams/pyrolite/pull/79/files) view) - once they're removed we can merge this in. Thanks for dealing with the merge conflict!

sarahshi commented:
I did not realize there were so, so many .DS_Store files. Apologies, I stop committing these files by adding to the gitignore file, but didn't for pyrolite. Let me know if that works!

morganjwilliams commented:
@sarahshi all good, it happens! 

I'll let the tests run then we can get this merged! 🥳

morganjwilliams commented:
@sarahshi it's in, many thanks for pulling this one together! 😄 

I'll look to add a few lines to the TAS documentation examples to highlight the new behavior. Also good time for a new release perhaps - I'll check out a few CIPW things and then look at v0.3.3.
